src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -208,6 +208,32 @@ public DocumentsOperationContext GetContextForNewDocument()
             public void Dispose()
             {
                 FinishBatchOfDocuments();
+                DeleteRevisionsForNonExistingDocuments();
+            }
+
+            private async void DeleteRevisionsForNonExistingDocuments()
+            {
+                if (_missingDocumentsForRevisions == null)
+                    return;
+                using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    foreach (var docId in _missingDocumentsForRevisions)
+                    {
+                        using (DocumentIdWorker.GetSliceFromId(context, docId.Key, out var lowerId))
+                        {
+                            var changeVector = _database.DocumentsStorage.GetNewChangeVector(context, _database.DocumentsStorage.GenerateNextEtag());
+                            var command = new RevisionsOperations.DeleteRevisionCommand(docId.Key,
```

Comment:
You need to pass the _missingDocumentsForRevisions into the command and process everything in one batch inside the command.

Reply:
Or batch the number of items by the same rules as we have in MergedBatchPutCommand.